crates/sparrow-physical/src/step.rs

```diff
@@ -90,3 +100,42 @@ pub enum StepKind {
     },
     Error,
 }
+
+impl StepKind {
+    /// Return true if the step is implemented as a transform (rather than an operation).
+    ///
+    /// This should generally be true for steps that take a single input and
+    /// effectively perform a "flat map" over the batches. Specifically, they
+    /// may change the number of rows in a batch (or omit the batch entirely)
+    /// and they may change the columns in the batch but they should not accept
+    /// multiple inputs or need to interact with scheduling.
+    ///
+    /// Examples:
+    ///
+    /// - `project` is a transform because it computes new columns for each row in
+    ///   every batch
+    /// - `filter` is a transform because it removes rows from each batch, and omits
+    ///   empty batches
+    /// - `merge` is not a transform because it accepts multiple inputs
+    /// - `shift` may be a transform if it uses the time in the input batch to determine
+    ///   which rows to output or it may be an operation if it interacts with scheduling
+    ///   in a more sophisticated way.
+    pub fn is_transform(&self) -> bool {
```

Comment:
I like the examples here. But the `shift` one is confusing coupled with this boolean method. `may be a transform` implies we'd need context to determine, but if we just call `is_transform` we'd get `false`. A caller may unknowingly call this without reading the docs and assume it's always not. 

(Though, I thought `shift` was never a `transform`)

Reply:
```
/// - `shift` depends on how we choose to implement it. If it is implemented as
///   a stateful transform that just buffers and emits as processing proceeds
///   through time, then it would be a transform. If we find ways to implement
///   it more efficiently by implementing the pipeline interface, then it may
///   not be a transform.
```